xray/commands/audit/applicabilitymanager.go

```diff
@@ -0,0 +1,355 @@
+// dependencies node ids.
+func getDirectDependenciesList(dependencyTrees []*services.GraphNode) []string {
+	directDependencies := []string{}
+	for _, tree := range dependencyTrees {
+		for _, node := range tree.Nodes {
+			directDependencies = append(directDependencies, node.Id)
+		}
+	}
+	return directDependencies
+}
+
+// Gets xray scan response and returns only the vulnerabilities part of it
+func getXrayVulnerabilities(xrayScanResults []services.ScanResponse) []services.Vulnerability {
+	xrayVulnerabilities := []services.Vulnerability{}
+	for _, result := range xrayScanResults {
+		xrayVulnerabilities = append(xrayVulnerabilities, result.Vulnerabilities...)
+	}
+	return xrayVulnerabilities
+}
+
+// Gets xray scan response and returns only the violations part of it
+func getXrayViolations(xrayScanResults []services.ScanResponse) []services.Violation {
+	xrayViolations := []services.Violation{}
+	for _, result := range xrayScanResults {
+		xrayViolations = append(xrayViolations, result.Violations...)
+	}
+	return xrayViolations
+}
+
+func (a *ApplicabilityScanManager) getApplicabilityScanResults() map[string]string {
+	return a.applicabilityScannerResults
+}
+
+func (a *ApplicabilityScanManager) Run() (bool, error) {
+	var err error
+	defer func() {
+		if a.deleteApplicabilityScanProcessFiles() != nil {
+			e := a.deleteApplicabilityScanProcessFiles()
+			if err == nil {
+				err = e
+			}
+		}
+	}()
+	if err = a.createConfigFile(); err != nil {
+		return true, err
+	}
+	if entitledForJas, err := a.runAnalyzerManager(); err != nil {
+		return entitledForJas, err
+	}
+	if err = a.parseResults(); err != nil {
+		return true, err
+	}
+	return true, nil
+}
+
+type applicabilityScanConfig struct {
+	Scans []scanConfiguration `yaml:"scans"`
+}
+
+type scanConfiguration struct {
+	Roots          []string `yaml:"roots"`
+	Output         string   `yaml:"output"`
+	Type           string   `yaml:"type"`
+	GrepDisable    bool     `yaml:"grep-disable"`
+	CveWhitelist   []string `yaml:"cve-whitelist"`
+	SkippedFolders []string `yaml:"skipped-folders"`
+}
+
+func (a *ApplicabilityScanManager) createConfigFile() error {
+	currentDir, err := coreutils.GetWorkingDirectory()
+	if err != nil {
+		return err
+	}
+	cveWhiteList := utils.RemoveDuplicateValues(createCveList(a.xrayVulnerabilities, a.xrayViolations))
+	configFileContent := applicabilityScanConfig{
+		Scans: []scanConfiguration{
+			{
+				Roots:          []string{currentDir},
+				Output:         a.resultsFileName,
+				Type:           applicabilityScanType,
+				GrepDisable:    false,
+				CveWhitelist:   cveWhiteList,
+				SkippedFolders: []string{},
+			},
+		},
+	}
+	yamlData, err := yaml.Marshal(&configFileContent)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(a.configFileName, yamlData, 0644)
+	return err
+}
+
+func (a *ApplicabilityScanManager) runAnalyzerManager() (bool, error) {
+	err := utils.SetAnalyzerManagerEnvVariables(a.serverDetails)
+	if errorutils.CheckError(err) != nil {
```

Comment:
The `CheckError ` should no be here, but inside the `SetAnalyzerManagerEnvVariables`. This is because our  convention is to pur the `CheckError` as early as downstream as possible as possible in the code. Not doing so misses the point of allowing us to see the stacktrace.